include/swift/AST/DiagnosticsSIL.def

```diff
@@ -173,6 +173,9 @@ ERROR(immutable_value_passed_inout,none,
 ERROR(assignment_to_immutable_value,none,
       "immutable value '%0' may not be assigned to",
       (StringRef))
+ERROR(mutating_protocol_witness_method_on_let_constant,none,
+      "cannot perform mutating instruction: '%0' is a 'let' constant",
```

Comment:
About consistency: I've checked again, what the error message is when the same issue can be caught on the AST level. It has been a while since I've last seen them. Sorry that I didn't have them on my mind when writing yesterdays comment.

<img width="913" alt="screen shot 2016-09-01 at 21 00 15" src="https://cloud.githubusercontent.com/assets/4222754/18180479/4bec20a4-7087-11e6-91e1-84d1f84d1304.png">

I do agree, that 'instruction' is the wrong term. While certainly correct on the SIL level, as a language user I personally wouldn't think of lines 27 & 28 as method invocations on `test`.

What do you think about `cannot perform mutating operation: '%0' is a 'let' constant` or `cannot use mutating operation with immutable value: '%0' is a 'let' constant`?